walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,137 @@
+        for (String person: attendees) {
+            // iterate through the events already scheduled for that day
+            for (Event mtg: events) {
+                Set<String> curAttending = mtg.getAttendees();
+                TimeRange curWhen = mtg.getWhen();
+                if (curAttending.contains(person)) {
+                    List<TimeRange> times = new ArrayList<>();
+                    times.addAll(newTimes);
+                    // iterate through the available times
+                    for (TimeRange time: times) {
+                        if (time.contains(curWhen)) {
+                            newTimes.remove(time);
+                            newTimes.add(TimeRange.fromStartEnd(time.start(), curWhen.start(), false));
+                            newTimes.add(TimeRange.fromStartEnd(curWhen.end(), time.end(), false));
+                        } else if (time.overlaps(curWhen)) {
+                            if (time.contains(curWhen.start())) {
+                                newTimes.remove(time);
+                                newTimes.add(TimeRange.fromStartEnd(time.start(), curWhen.start(), false));
+                            } else if (curWhen.contains(time.start())) {
+                                newTimes.remove(time);
+                                newTimes.add(TimeRange.fromStartEnd(curWhen.end(), time.end(), false));
+                            }
+                        }
+                        if (curWhen.contains(time)) {
+                            newTimes.remove(time);
+                        }
+                    }
+                }
+            }
+        }
+        return newTimes;
+    }
+
+    public List<TimeRange> sorter(List<TimeRange> finalTimes, MeetingRequest request, Set<TimeRange> semiFilteredTimes) {
+        for (TimeRange time: semiFilteredTimes) {
+            if (time.duration() != 0 && request.getDuration() <= time.duration()) {
+                if (finalTimes.size() == 0) {
+                    finalTimes.add(time);
+                } else {
+                    for (int i = 0; i < finalTimes.size(); i++) {
+                        if (time.start() < finalTimes.get(i).start()) {
+                            finalTimes.add(i, time);
+                        } else if (time.start() == finalTimes.get(i).start()) {
+                            if (time.end() < finalTimes.get(i).end()) {
+                                finalTimes.add(i, time);
+                            } else {
+                                finalTimes.add(i + 1, time);
+                            }
+                        } else {
+                            finalTimes.add(time);
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+        return finalTimes;
+    }
+
+    public List<TimeRange> comparer(List<TimeRange> optTimes, List<TimeRange> newTimes, MeetingRequest request) {
```

Comment:
What if you sorted them first using one of the provided comparators and then merge the intervals?